marimo/_runtime/executor.py

```diff
@@ -114,28 +162,63 @@ async def execute_cell_async(
         if cell.body is None:
             return None
         assert cell.last_expr is not None
-
-        if _is_coroutine(cell.body):
-            await eval(cell.body, glbls)
-        else:
-            exec(cell.body, glbls)
-
-        if _is_coroutine(cell.last_expr):
-            return await eval(cell.last_expr, glbls)
-        else:
-            return eval(cell.last_expr, glbls)
+        try:
+            if _is_coroutine(cell.body):
+                # fmt: off
+                (
+                    await marimo_output_evaluation
+                    (cell.body, glbls)
+                )
+                # fmt: on
+            else:
+                # fmt: off
+                (
+                    marimo_body_execution
+                    (cell.body, glbls)
+                )
+                # fmt: on
+
+            if _is_coroutine(cell.last_expr):
+                # fmt: off
+                return (
+                    await marimo_output_evaluation
+                    (cell.last_expr, glbls)
+                )
+                # fmt: on
+            else:
+                # fmt: off
+                return (
+                    marimo_output_evaluation
+                    (cell.last_expr, glbls)
+                )
+                # fmt: on
+        except Exception as e:
```

Comment:
This should be `except BaseException as e:`. Some libraries throw `BaseExceptions` (polars is one), and we need to communicate that exception info back to the user instead of hiding it in an "Unknown Error".

Reply:
Taking a step back, related to my comment above — instead of handling a special `MarimoBaseException`, would it be possible to just rewrite the traceback and hide marimo internals?

Reply:
Could just modify the traceback by throwing out all frames preceding the marimo cell's execution (ie lines before `File  ...__marimo__cell...`)

Reply:
Good catch! This is the source of the failing test. Fixed. Re-stack trace manipulation, see other comment.

I think that `MarimoBaseException` is probably not correctly named. It's really the `MarimoExecutionException` or `CellRuntimeException`, and provides the context that it was a evaluation cell that failed. Being able to unwrap the exception for a cleaner stack trace is definitely a plus, but not the only positive since it's an indication the failure came from a cell failure.

LMK if this context changes your thoughts/ if you have thoughts for a better name